### ISSUE TITLE
[SNOWFLAKE] RF04 - 'queue' incorrectly flagged as keyword identifier in Snowflake dialect

### ISSUE THREAD

**Issue by joshuanits** [sqlfluff/sqlfluff#7089]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

SQLFluff raises RF04 - Keywords should not be used as identifiers for the `queue` identifier in the Snowflake dialect.

As per [Reserved & limited keywords | Snowflake Documentation](https://docs.snowflake.com/en/sql-reference/reserved-keywords), `queue` is not a reserved word in Snowflake.

### Expected Behaviour

SQLFluff doesn't raise RF04 for 'queue' as an identifier in Snowflake dialect.

### Observed Behaviour

SQLFluff returns the following error:

```
L:   1 | P:  13 | RF04 | Keywords should not be used as identifiers.
                       | [references.keywords]
```

### How to reproduce

Steps to reproduce:
```bash
echo 'select 1 as queue' > test.sql
sqlfluff lint test.sql --dialect snowflake
```

Output:
```bash
== [test.sql] FAIL
L:   1 | P:  13 | RF04 | Keywords should not be used as identifiers.
                       | [references.keywords]
All Finished 📜 🎉!
```

### Dialect

snowflake

### Version

sqlfluff, version 3.4.2

### Configuration

none

### Are you willing to work on and submit a PR to address the issue?

- [x] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by joshuanits** [comment]
I see that it's (correctly) flagged as an unreserved keyword in the source. Is there no configuration option to ignore unreserved keywords?

It seems like more of an anti-pattern to rename identifiers just to avoid clashes with an unreserved keyword like queue that is seldom used.

**Comment by ifm-anowaczek** [comment]
> Is there no configuration option to ignore unreserved keywords?

There is `unquoted_identifiers_policy` which allows you to choose which identifiers to fail and it has the following options: `['all', 'aliases', 'column_aliases']`. So you can't set it to `none` like you can with the `quoted_identifiers_policy` configuration option.

You would think this means the solution is to just quote the identifier (as opposed to renaming it). But there is an issue where RF06 will unquote your identifier because it thinks the unquoted identifier will work fine (technically true for the actual SQL, but then it will fail RF04 again). I have an open issue about this: #7072 

For now, you can use an inline ignore (`--noqa: RF04`) or populate the `ignore_words` configuration option for RF04 with the keywords you want to use.

**Comment by WittierDinosaur** [comment]
@ifm-anowaczek covers the solutions. I think there's nothing to fix here and it's working as intended, so i'm gonna close this issue. You can also just turn the rule off if it's not one you want to implement
